### PR TITLE
[RDS] Add ``data/opentelekomcloud_rds_instance_v3``

### PR DESCRIPTION
**Pull request by vladimirvshivkov**
## PR Checklist

* [x] Refers to: #1736
* [x] Tests added/passed.
* [x] Schema updated.
* [x] Release notes added.
* [x] Documentation added. 

## Acceptance Steps Performed

```
=== RUN   TestAccRdsInstanceDataSource_basic
=== PAUSE TestAccRdsInstanceDataSource_basic
=== CONT  TestAccRdsInstanceDataSource_basic
--- PASS: TestAccRdsInstanceDataSource_basic (712.50s)
PASS

Process finished with exit code 0
```

### CODE REVIEW

File: opentelekomcloud/services/rds/data_source_opentelekomcloud_rds_instances_v3.go [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+				Optional: true,
+			},
+			"subnet_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"enterprise_project_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
```
Comment (lego963) [inline opentelekomcloud/services/rds/data_source_opentelekomcloud_rds_instances_v3.go]: this is unsupported in the OTC

File: go.mod [hunk 2]
```diff
@@ -4,7 +4,7 @@ go 1.17
 
 require (
 	github.com/apparentlymart/go-cidr v1.1.0
-	github.com/aws/aws-sdk-go v1.36.29
+	github.com/aws/aws-sdk-go v1.37.0
```
Comment (lego963) [inline go.mod:7]: do we need it?

File: opentelekomcloud/acceptance/rds/data_source_opentelekomcloud_rds_instances_v3_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+resource "opentelekomcloud_vpc_subnet_v1" "test" {
+  name       = "%s"
+  cidr       = "192.168.0.0/24"
+  gateway_ip = "192.168.0.1"
+  vpc_id     = opentelekomcloud_vpc_v1.test.id
+}
+
+resource "opentelekomcloud_networking_secgroup_v2" "test" {
+  name = "%s"
+}
```
Comment (lego963) [inline opentelekomcloud/acceptance/rds/data_source_opentelekomcloud_rds_instances_v3_test.go]: We don't need to create network infra, can you use predefined infra like in `rds_instance_v3` test?

- `common.DataSourceSecGroupDefault`
- `common.DataSourceSubnet`

File: opentelekomcloud/services/rds/data_source_opentelekomcloud_rds_instances_v3.go [hunk 4]
```diff
@@ -0,0 +1,378 @@
+package rds
```
Comment (lego963) [inline opentelekomcloud/services/rds/data_source_opentelekomcloud_rds_instances_v3.go]: opentelekomcloud/services/rds/data_source_opentelekomcloud_rds_instances_v3.go
should be
opentelekomcloud/services/rds/data_source_opentelekomcloud_rds_instance_v3.go

File: opentelekomcloud/services/rds/data_source_opentelekomcloud_rds_instances_v3.go [hunk 5]
```diff
@@ -0,0 +1,378 @@
...
+							Computed: true,
+						},
+						"security_group_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"enterprise_project_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
```
Comment (lego963) [inline opentelekomcloud/services/rds/data_source_opentelekomcloud_rds_instances_v3.go]: same

File: opentelekomcloud/services/rds/data_source_opentelekomcloud_rds_instances_v3.go [hunk 6]
```diff
@@ -0,0 +1,378 @@
...
+}
+
+func dataSourceRdsInstanceV3Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*cfg.Config)
+	client, err := config.RdsV3Client(config.GetRegion(d))
+	if err != nil {
+		return fmterr.Errorf(errCreateClient, err)
+	}
+
+	listOpts := instances.ListOpts{
```
Comment (lego963) [inline opentelekomcloud/services/rds/data_source_opentelekomcloud_rds_instances_v3.go]: you missed `id` filter

File: opentelekomcloud/services/rds/data_source_opentelekomcloud_rds_instances_v3.go [hunk 7]
```diff
@@ -0,0 +1,378 @@
...
+	}
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	var instancesToSet []map[string]interface{}
+	var instancesIds []string
+
+	for _, item := range allInstances.Instances {
```
Comment (lego963) [inline opentelekomcloud/services/rds/data_source_opentelekomcloud_rds_instances_v3.go]: in `data_source/opentelekomcloud_rds_instance_v3` we should return only single instance

```
	if len(allInstances) < 1 {
		return common.DataSourceTooFewDiag
	}

	if len(allInstances) > 1 {
		return common.DataSourceTooManyDiag
	}
```

**Review by lego963 — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (vladimirvshivkov) [reply go.mod:7]: go mod thinks we do

File: opentelekomcloud/services/rds/data_source_opentelekomcloud_rds_instance_v3.go [hunk 8]
```diff
@@ -0,0 +1,378 @@
...
+
+	if len(allInstances.Instances) < 1 {
+		return common.DataSourceTooFewDiag
+	}
+
+	if len(allInstances.Instances) > 1 {
+		return common.DataSourceTooManyDiag
+	}
+
+	for _, item := range allInstances.Instances {
```
Comment (lego963) [inline opentelekomcloud/services/rds/data_source_opentelekomcloud_rds_instance_v3.go]: you don't need to use `for` if you have 1 instance

File: docs/data-sources/rds_instance_v3.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
+---
+subcategory: "Relational Database Service (RDS)"
+---
+
+# opentelekomcloud_rds_instance_v3
+
```
Comment (lego963) [inline docs/data-sources/rds_instance_v3.md:6]: add small description here please

Comment (vladimirvshivkov) [reply docs/data-sources/rds_instance_v3.md:6]: done

**Review by lego963 — APPROVED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Pull request merged**